client.go

```diff
@@ -69,43 +67,31 @@ func (c *Client) GetBaseUrl() string {
 
 // Validate checks if the API and application keys are valid.
 func (client *Client) Validate() (bool, error) {
-	var bodyreader io.Reader
 	var out valid
+	var resp *http.Response
+
 	uri, err := client.uriForAPI("/v1/validate")
 	if err != nil {
 		return false, err
 	}
-	req, err := http.NewRequest("GET", uri, bodyreader)
 
-	if err != nil {
-		return false, err
-	}
-	if bodyreader != nil {
-		req.Header.Add("Content-Type", "application/json")
-	}
+	req, err := http.NewRequest("GET", uri, nil)
```

Comment:
Still needs checking if we got an error here.

@GrantSheehan btw, you were absolutely right about inline error checking. I stand corrected!